Rarely, title-bar may not be resized on snapping to bottom/top by drag-and-drop

It happened here very randomly and rarely when I snapped a window to the bottom of the screen by dragging its title-bar and dropping it to the bottom edge. Fortunately, I took a screenshot:

![bottom](https://github.com/labwc/labwc/assets/981076/0109ad9f-85ab-4d08-aefa-bd7fd7a24685)

I encountered it only once with the top edge and haven't seen it without drag-and-drop. However, I can't be sure that it's really related to drag-and-drop.

Sorry if it's already known or fixed in git; I use 0.6.5 and couldn't find it after a simple search.

I think our code is correct (using the window geometry is the right thing to do, in principle). I had something like this in mind for a workaround:
https://github.com/jlindgren90/labwc/commit/7a69cd2ed8a7a5b5787c7061193672e5658d1a47

In @Flrian's log, the commit did happen but was just a hair too late (116ms):
```
00:00:05.482 [ERROR] [../src/xdg.c:301] view 0x561ca4b66730 (calibre — || Calibre Library ||) current=680x498 pending=680x498 configure=1898x498
00:00:05.482 [ERROR] [../src/xdg.c:308] configure serial=26
00:00:05.482 [ERROR] [../src/xdg.c:149] set pending serial=26
00:00:05.482 [DEBUG] [types/output/cursor.c:432] Falling back to software cursor on output 'HDMI-A-1'
00:00:05.582 [INFO] [../src/xdg.c:127] client (calibre-gui) did not respond to configure request in 100 ms
00:00:05.582 [ERROR] [../src/xdg.c:129] view 0x561ca4b66730 (calibre — || Calibre Library ||) current=680x498 pending=1898x498 at timeout
00:00:05.582 [ERROR] [../src/xdg.c:133] pending serial=26
00:00:05.598 [ERROR] [../src/xdg.c:92] view 0x561ca4b66730 (calibre — || Calibre Library ||) current=680x498 pending=1898x498 commit=680x498
00:00:05.598 [ERROR] [../src/xdg.c:97] pending serial=0 commit=26
```

--- a/src/xdg.c
+++ b/src/xdg.c
@@ -90,6 +90,27 @@ handle_commit(struct wl_listener *listener, void *data)
 	struct wlr_box size;
 	wlr_xdg_surface_get_geometry(xdg_surface, &size);
 
+	/*
+	 * Qt applications occasionally fail to call set_window_geometry
+	 * after a configure request, but do correctly update the actual
+	 * surface extent. This results in a mismatch between the window
+	 * decorations (which follow the logical geometry) and the visual
+	 * size of the client area. As a workaround, we try to detect
+	 * this case and ignore the out-of-date window geometry.
+	 */
+	if (size.width != view->pending.width
+			|| size.height != view->pending.height) {
+		struct wlr_box extent;
+		wlr_surface_get_extends(xdg_surface->surface, &extent);
+		if (extent.width == view->pending.width
+				&& extent.height == view->pending.height) {
+			wlr_log(WLR_DEBUG, "window geometry for client (%s) "
+				"appears to be incorrect - ignoring",
+				view_get_string_prop(view, "app_id"));
+			size = extent; /* Use surface extent instead */
+		}
+	}
+
 	struct wlr_box *current = &view->current;
 	bool update_required = current->width != size.width
 		|| current->height != size.height;

